Source/WebCore/Modules/webcodecs/WebCodecsVideoEncoder.cpp

```diff
@@ -123,13 +129,37 @@ ExceptionOr<void> WebCodecsVideoEncoder::configure(WebCodecsVideoEncoderConfig&&
                 result.duration,
                 BufferSource { WTFMove(buffer) }
             });
-            // FIXME: Implement metadata.
-            m_output->handleEvent(WTFMove(chunk), { });
+            m_output->handleEvent(WTFMove(chunk), createEncodedChunkMetadata());
         }, WTFMove(postTaskCallback));
     });
     return { };
 }
 
+WebCodecsEncodedVideoChunkMetadata WebCodecsVideoEncoder::createEncodedChunkMetadata()
+{
+    WebCodecsVideoDecoderConfig decoderConfig;
+    if (!m_hasNewActiveConfiguration)
+        return { };
+
+    m_hasNewActiveConfiguration = false;
+    WebCodecsVideoDecoderConfig config {
+        WTFMove(m_activeConfiguration.codec),
+        { },
+        m_activeConfiguration.visibleWidth,
+        m_activeConfiguration.visibleHeight,
+        m_activeConfiguration.displayWidth,
+        m_activeConfiguration.displayHeight
+    };
+    if (m_activeConfiguration.description) {
+        if (auto arrayBuffer = ArrayBuffer::tryCreateUninitialized(m_activeConfiguration.description->size(), 0)) {
+            memcpy(static_cast<uint8_t*>(arrayBuffer->data()), m_activeConfiguration.description->data(), m_activeConfiguration.description->size());
+            config.description = WTFMove(arrayBuffer);
+        }
```

Comment:
Might be worth logging if `ArrayBuffer::tryCreateUninitialized` fails

Reply:
Will do.